embedded/src/main/java/org/wildfly/core/embedded/EmbeddedStandaloneServerFactory.java

```diff
@@ -266,8 +267,12 @@ public void exit(int status) {
                     });
 
                     // Determine the ServerEnvironment
-                    ServerEnvironment serverEnvironment = Main.determineEnvironment(cmdargs, systemProps, systemEnv, ServerEnvironment.LaunchType.EMBEDDED, startTime).getServerEnvironment();
-
+                    ServerEnvironmentWrapper wrapper = Main.determineEnvironment(cmdargs, systemProps, systemEnv, ServerEnvironment.LaunchType.EMBEDDED, startTime);
+                    ServerEnvironment serverEnvironment = wrapper.getServerEnvironment();
+                    if (serverEnvironment == null && ServerEnvironmentWrapper.ServerEnvironmentStatus.NORMAL.equals(wrapper.getServerEnvironmentStatus())) {
+                        // Nothing to do
+                        return;
```

Comment:
what would happen in the wrapper's environment status is `ERROR`?

Reply:
@jmesnil , I overlooked the Error case. We would get an NPE in case of ERROR. There is no need to check for the status actually. A null env is enough to return.